lib/tier-node.js

```diff
@@ -89,9 +89,24 @@ function handler (handle, params, req, res, body, query) {
 
   const request = new Request(params, req, body, query)
   const reply = new Reply(req, res, handle)
-  const result = handle.handler(request, reply)
-  if (result && typeof result.then === 'function') {
-    reply.send(result)
+
+  mapSeries(onRequest(), _onRequestIterator, _runOnRequestWrap(request, reply, handle))
+
+  function _onRequestIterator (fn, cb) {
+    setImmediate(fn, request, reply, cb)
+  }
+}
+
+function _runOnRequestWrap (request, reply, handle) {
+  return function _runOnRequest (err) {
+    if (err) {
+      reply.send(err)
+      return
+    }
+    const result = handle.handler(request, reply)
+    if (result && typeof result.then === 'function') {
+      reply.send(result)
+    }
```

Comment:
same here, see https://github.com/mcollina/aedes/blob/master/lib/handlers/subscribe.js#L25-L29 to avoid creating closures.